javascript/javascript1/week4/script.js

```diff
@@ -0,0 +1,135 @@
+    } else {
+        return `${userName.charAt(0).toUpperCase() + userName.slice(1)}`;
+    }
+ }
+
+function addTodo(todo) {
+    
+    if (todoList.includes(todo)) {
+        return 'Todo is already in the list';
+    } else {
+        todoList.push(todo);
+        return `${todo} added to your todo`;
+    }
+}
+
+function removeTodo (todo) {
+
+    if (todoList.length === 0) {
+        return 'Your todo list is empty!'
+    } else if (!todoList.includes(todo)) {
+        return "The todo doesn't exist in your todo list";
+    } else if (todoList.includes(todo)) {
+        const index = todoList.indexOf(todo);
+        todoList.splice(index, 1);
+        return `Removed ${todo} from your todo`;
+    }
+}
+
+function getTodo() {
+    
+    if (todoList.length === 0) {
+        return "You todo is empty"
+    }
+    return todoList;
+}
+
+function getDate() {
+        const date = new Date();
+        const day = date.getDate();
+        const month = date.toLocaleString("en", { month: "long"  });
+        const year = date.getFullYear();
+        
+        return ` ${day} of ${month} ${year}`;
+}
+
+function calculation(command) {
+    
+        const action = command.toLowerCase().slice(7);
+        return eval(action);
+    }
+
+function setTimer(command) {
+    
+        const ms = command.toLowerCase().split(' ').splice(4, 1) * 10000 ;
+        console.log(`Timer set for ${ms} minutes`);
+        
+        return  setTimeout(() => {
+            console.log("Timer done");
+        }, ms);
+}
+
+
+function getReply(command) {
+
+    // command transformation into an array of keywords
+    const stringWithoutSpecChars = command.replace(/[^\w\s]/gi, '')
+    const commandTransformToArray = stringWithoutSpecChars.toLowerCase().split(' ');
+    const arrayWithoutDuplicatedWords = commandTransformToArray.filter((word, index, array) => array.indexOf(word) === index);   //    make new array without dabbled words
+    
+    const todoItem = commandTransformToArray.slice(1, - 3);
+    const todo = todoItem.join(' ');
+
+    
+    // check conditions if a keyword has matches to run a corresponding callback function with an array of non-duplicated words 
+    if (arrayWithoutDuplicatedWords.includes('name') && arrayWithoutDuplicatedWords[0] === 'hello') {
+        return setName(arrayWithoutDuplicatedWords);
+    }
+    
+    if (arrayWithoutDuplicatedWords.includes('what') && arrayWithoutDuplicatedWords.includes('name')) {
+        return getName(arrayWithoutDuplicatedWords);
+    }
+    
+    if (arrayWithoutDuplicatedWords[0] === 'add' && arrayWithoutDuplicatedWords.includes('todo') ) {
+        return addTodo(todo);
+    }
+    
+    if (arrayWithoutDuplicatedWords.includes('remove') && arrayWithoutDuplicatedWords.includes('todo')) {
+         return removeTodo(todo);
+        }
+        
+    if (arrayWithoutDuplicatedWords.includes('what') && arrayWithoutDuplicatedWords.includes('todo')) {
+        return  getTodo();
+    }
+    
```

Comment:
this console.log(getReply('what fishing from my todo')); 
breaks your code :)
can you think of ways to make your code safe for such ..uninformed users?

Reply:
done )